### DISCUSSION TITLE
Text prompts - segment-geospatial

### DISCUSSION THREAD

**Discussion by arabinda90** [opengeos/segment-geospatial#195] [Q&A] [answered]
Dear sir,
Sir, when I apply it to your sample data it is well executed but, when I it apply it to Sentinel-2 data it shows an error. Can you give me a solution?

#--------------------------------------------------------------------------------------
sam.predict(image, text_prompt, box_threshold=0.24, text_threshold=0.24)
#---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
File ~\anaconda3\envs\gdal_env\lib\site-packages\PIL\Image.py:2953, in fromarray(obj, mode)
   2952 try:
-> 2953     mode, rawmode = _fromarray_typemap[typekey]
   2954 except KeyError as e:

KeyError: ((1, 1, 3), '<f4')

The above exception was the direct cause of the following exception:

TypeError                                 Traceback (most recent call last)
Input In [7], in <cell line: 1>()
----> 1 sam.predict(image, text_prompt, box_threshold=0.24, text_threshold=0.24)

File ~\anaconda3\envs\gdal_env\lib\site-packages\samgeo\text_sam.py:273, in LangSAM.predict(self, image, text_prompt, box_threshold, text_threshold, output, mask_multiplier, dtype, save_args, return_results, return_coords, **kwargs)
    271         self.transform = src.transform  # Save georeferencing information
    272         self.crs = src.crs  # Save the Coordinate Reference System
--> 273         image_pil = Image.fromarray(
    274             image_np[:, :, :3]
    275         )  # Convert numpy array to PIL image, excluding the alpha channel
    276 else:
    277     image_pil = image

File ~\anaconda3\envs\gdal_env\lib\site-packages\PIL\Image.py:2955, in fromarray(obj, mode)
   2953         mode, rawmode = _fromarray_typemap[typekey]
   2954     except KeyError as e:
-> 2955         raise TypeError("Cannot handle this data type: %s, %s" % typekey) from e
   2956 else:
   2957     rawmode = mode

TypeError: Cannot handle this data type: (1, 1, 3), <f4

**Comment by giswqs** [comment, marked as answer]
For now, samgeo only support images with RGB and 8-bit integer type.

**Comment by arabinda90** [comment]
Thank you, sir, for your assistance.

So if I converted 32 bits RGB band of Sentinel-2  data to 8 bit then it will work?
